poncho/src/poncho/package_serverize.py

```diff
@@ -14,226 +14,221 @@
+        return remote_wrapper
+        
+    read, write = os.pipe() 
 
-def send_configuration(config):
-    config_string = json.dumps(config)
-    config_cmd = f"{len(config_string) + 1}\\n{config_string}\\n"
-    sys.stdout.write(config_cmd)
-    sys.stdout.flush()
+    def send_configuration(config):
+        config_string = json.dumps(config)
+        config_cmd = f"{len(config_string) + 1}\n{config_string}\n"
+        sys.stdout.write(config_cmd)
+        sys.stdout.flush()
 
-def main():
-    config = {
-        "name": name(),
-    }
-    send_configuration(config)
-    while True:
+    def main():
+        config = {
+            "name": name(),
+        }
+        send_configuration(config)
         while True:
-            # wait for message from worker about what function to execute
-            line = input()
-            print(f"Network function received task", file=sys.stderr, flush=True)
-            if len(line) >= 0:
-                function_name, event_size = line.split(" ")
-                if event_size:
-                    # receive the bytes containing the event and turn it into a string
-                    event_str = input()
-                    if len(event_str) != int(event_size):
-                        print(event_str, len(event_str), event_size, file=sys.stderr)
-                        print("Size of event does not match what was sent: exiting", file=sys.stderr)
-                        exit(0)
-                    # turn the event into a python dictionary
-                    event = json.loads(event_str)
-                    # see if the user specified an execution method
-                    exec_method = event.get("remote_task_exec_method", None)
-                    if exec_method == "thread":
-                        # create a forked process for function handler
-                        q = queue.Queue()
-                        p = threading.Thread(target=globals()[function_name], args=(event_str, q))
-                        p.start()
-                        p.join()
-                        response = json.dumps(q.get())
-                    elif exec_method == "direct":
-                        response = json.dumps(globals()[function_name](event))
-                    else:
-                        p = os.fork()
-                        if p == 0:
-                            response =globals()[function_name](event)
-                            os.write(write, json.dumps(response).encode("utf-8"))
-                            os._exit(0)
-                        elif p < 0:
-                            print('Network function: unable to fork', file=sys.stderr)
-                            response = { 
-                                "Result": "unable to fork",
-                                "StatusCode": 500 
-                            }
+            while True:
+                # wait for message from worker about what function to execute
+                line = input()
```

Comment:
try ... except EOFError  wrap around this line. On EOF, probably break out of the loop?

Reply:
I made the process exit on EOF. The only case where EOF should happen is if the worker closes the write end of the pipe, and in that case the coprocess can't do much of use so it should just terminate itself.